src/Sulu/Bundle/MediaBundle/Markup/MediaTag.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\MediaBundle\Markup;
+
+use Sulu\Bundle\MarkupBundle\Tag\TagInterface;
+use Sulu\Bundle\MediaBundle\Entity\Media;
+use Sulu\Bundle\MediaBundle\Media\Manager\MediaManagerInterface;
+use Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactory;
+use Sulu\Component\Rest\ListBuilder\Metadata\FieldDescriptorFactoryInterface;
+
+/**
+ * Extends the sulu markup with the "sulu:link" tag.
+ */
+class MediaTag implements TagInterface
+{
+    const VALIDATE_REMOVED = 'removed';
+
+    /**
+     * @var string
+     */
+    protected static $entityName = 'SuluMediaBundle:Media';
+
+    /**
+     * @var DoctrineListBuilderFactory
+     */
+    private $listBuilderFactory;
+
+    /**
+     * @var FieldDescriptorFactoryInterface
+     */
+    private $fieldDescriptorFactory;
+
+    /**
+     * @var MediaManagerInterface
+     */
+    private $mediaManager;
+
+    /**
+     * @param DoctrineListBuilderFactory $listBuilderFactory
+     * @param FieldDescriptorFactoryInterface $fieldDescriptorFactory
+     * @param MediaManagerInterface $mediaManager
+     */
+    public function __construct(
+        DoctrineListBuilderFactory $listBuilderFactory,
+        FieldDescriptorFactoryInterface $fieldDescriptorFactory,
+        MediaManagerInterface $mediaManager
+    ) {
+        $this->listBuilderFactory = $listBuilderFactory;
+        $this->fieldDescriptorFactory = $fieldDescriptorFactory;
+        $this->mediaManager = $mediaManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parseAll(array $attributesByTag, $locale)
+    {
+        $medias = $this->preloadMedias($attributesByTag, $locale);
+
+        $result = [];
+        foreach ($attributesByTag as $tag => $attributes) {
+            if (!array_key_exists($attributes['id'], $medias)) {
+                $result[$tag] = array_key_exists('content', $attributes) ? $attributes['content'] :
```

Comment:
Can we make something like a "createMarkupForAttributes" method and outsource to body of this foreach?